packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,135 +5,131 @@
-  const {
-    themeConfig: {algolia},
-  } = siteConfig;
-  const history = useHistory();
-  const {navigateToSearchPage} = useSearchQuery();
-
-  function initAlgolia(focus) {
-    window.docsearch({
-      appId: algolia.appId,
-      apiKey: algolia.apiKey,
-      indexName: algolia.indexName,
-      inputSelector: '#search_input_react',
-      algoliaOptions: algolia.algoliaOptions,
-      autocompleteOptions: {
-        openOnFocus: true,
-        autoselect: false,
-        hint: false,
-        tabAutocomplete: false,
-      },
-      // Override algolia's default selection event, allowing us to do client-side
-      // navigation and avoiding a full page refresh.
-      handleSelected: (_input, _event, suggestion) => {
-        _event.stopPropagation();
-
-        // Use an anchor tag to parse the absolute url into a relative url
-        // Alternatively, we can use new URL(suggestion.url) but it's not supported in IE.
-        const a = document.createElement('a');
-        a.href = suggestion.url;
-
-        // Algolia use closest parent element id #__docusaurus when a h1 page title does
-        // not have an id, so we can safely remove it.
-        // See https://github.com/facebook/docusaurus/issues/1828 for more details.
-        const routePath =
-          `#__docusaurus` === a.hash
-            ? `${a.pathname}`
-            : `${a.pathname}${a.hash}`;
-        history.push(routePath);
-      },
-    });
+let DocSearchModal = null;
 
-    if (focus) {
-      searchBarRef.current.focus();
-    }
-  }
+function Hit({hit, children}) {
+  return <Link to={hit.url}>{children}</Link>;
+}
 
-  const loadAlgolia = (focus = true) => {
-    if (algoliaLoaded) {
-      return;
-    }
+function ResultsFooter({state}) {
+  const {generateSearchPageLink} = useSearchQuery();
 
-    Promise.all([import('docsearch.js'), import('./algolia.css')]).then(
-      ([{default: docsearch}]) => {
-        setAlgoliaLoaded(true);
-        window.docsearch = docsearch;
-        initAlgolia(focus);
-      },
-    );
-  };
+  return (
+    <Link to={generateSearchPageLink(state.query)}>
+      See all {state.context.nbHits} results
+    </Link>
+  );
+}
 
-  const toggleSearchInput = useCallback(() => {
-    loadAlgolia();
+function DocSearch({indexName, appId, apiKey, searchParameters}) {
+  const history = useHistory();
+  const [isOpen, setIsOpen] = useState(false);
 
-    if (algoliaLoaded) {
-      searchBarRef.current.focus();
+  const importDocSearchModalIfNeeded = useCallback(() => {
+    if (DocSearchModal) {
+      return Promise.resolve();
     }
 
-    handleSearchBarToggle(!isSearchBarExpanded);
-  }, [isSearchBarExpanded]);
-
-  const handleSearchInputBlur = useCallback(() => {
-    handleSearchBarToggle(!isSearchBarExpanded);
-  }, [isSearchBarExpanded]);
+    return Promise.all([
+      import('@docsearch/react/modal'),
+      import('@docsearch/react/style'),
+      import('./styles.css'),
```

Comment:
hey, wondering if we couldn't add Webpack prefetch hints here?

https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules

So that the bundles might be already loaded as a low-priority request, and first user click does not need to wait for fetches?

Reply:
Sure, I've never used it but it seems like a good fit. I'll try to make these imports static with the prefetch hint.

Reply:
Is that what you had in mind?

```js
function DocSearch() {
  // ...

  const importDocSearchModalIfNeeded = useCallback(() => {
    if (DocSearchModal) {
      return Promise.resolve();
    }

    return Promise.all([
      import(/* webpackPrefetch: true */ '@docsearch/react/modal'),
      import(/* webpackPrefetch: true */ '@docsearch/react/style'),
      import(/* webpackPrefetch: true */ './styles.css'),
    ]).then(([{DocSearchModal: Modal}]) => {
      DocSearchModal = Modal;
    });
  }, []);

  React.useEffect(() => {
    importDocSearchModalIfNeeded();
  }, []);

  // ...
}
```

Reply:
sorry, missed your comment.

yes this is what I had in mind, but nevermind, for now we don't have any prefetch hint anyway, so we'll see later if this can be useful